### PR TITLE
Explain the default for TLSConfig.mode in documentation

### LINKED ISSUE #2070: TLSConfig.mode is "optional", but has no default
**What happened**:
Looking at https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.GatewayTLSConfig, the `mode` is optional, but it is not explained what it means if this is missing.

The `certificateRefs` field says:

> This field is required to have at least one element when the mode is set to “Terminate” (default) and is optional otherwise.

Which suggests that "terminate" is actually the default.

**What you expected to happen**:
It is clearly stated what happens if the field isn't present.

**Anything else we need to know?**:
Neither [Envoy Gateway](https://github.com/envoyproxy/gateway/blob/80db45944e010a5ba8eb2636201450ffb1a59f8e/internal/provider/kubernetes/helpers.go#L182) nor [cert-manager](https://github.com/cert-manager/cert-manager/blob/c5e6bf39d688d202592318eaf91988466a7ee37b/pkg/controller/certificate-shim/sync.go#L283) handle "terminate" as the default.

### PR DESCRIPTION
**Pull request by tommie**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Clarifies that TLSConfig.mode has a default.

It seems the doc generator doesn't look at the `kubebuilder:default` tag, though it could.

**Which issue(s) this PR fixes**:

Fixes #2070 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: apis/v1beta1/gateway_types.go [hunk 1]
```diff
@@ -338,7 +338,7 @@ type GatewayTLSConfig struct {
 	// Mode defines the TLS behavior for the TLS session initiated by the client.
 	// There are two possible modes:
 	//
-	// - Terminate: The TLS session between the downstream client
+	// - Terminate (the default): The TLS session between the downstream client
```
Comment (danehans) [inline apis/v1beta1/gateway_types.go:341]: nit: s/the default/default/

**Comment by robscott** [comment]
Thanks @tommie! This LGTM once the suggestion from @danehans is included.

/ok-to-test

**Review by arkodg — APPROVED** [review]
/approve

**Comment by youngnick** [comment]
I think this is fine, I'll give this one more day, then LGTM. Ping @robscott and @shaneutt in case they want to take another look.

**Comment by youngnick** [comment]
Oh, sorry, I missed this @tommie. Could you please rebase and then I'll merge?

**Comment by tommie** [comment]
> Oh, sorry, I missed this @tommie. Could you please rebase and then I'll merge?

@youngnick Yepp, done.

**Comment by youngnick** [comment]
Sorry @tommie, I tried to sort this out in the Github UI, but looks like it needs a gofumpt run. Do you want to do it or should I?

**Comment by mlavacca** [comment]
Hi @tommie, we are going through the PRs that have been inactive for a while, as our bot configuration is configured in a pretty loose way and we are trying to perform some cleanup in the open issues. We really appreciate your effort here, but since this PR has been in stale for a while now, we are closing it.

In case you have an interest, feel free to re-open the PR, rebase and we'll merge it.

/close

**Pull request closed without merging**